### ISSUE TITLE
Selecting text with blinking cursor

### ISSUE THREAD

**Issue by nunocastromartins** [jeremyramin/terminal-plus#67]
I believe the title says it all. Selecting text with the "Cursor Blink" option enabled is impossible as the text becomes unselected when a "blink" occurs.

This is troublesome when attempting to copy text from the terminal, for instance.

**Comment by jeremyramin** [comment]
@nunocastromartins I'm assuming you mean when trying to copy text from where the prompt is. This issue is a side effect of my dependency trying to animate within HTML. 

The dependency animates the cursor blink by editing the HTML at that line to remove and add the cursor. When that happens, text selected at that region is deselected because the text disappeared for a moment in time.

I might be able to fix this. I'll edit the dependency and try a different way of animating that could solve this issue.

**Comment by nunocastromartins** [comment]
You can do this with CSS only. :smile:
Give me a moment and I'll send you something to work with.

**Comment by jeremyramin** [comment]
I figured I could just set up a blinking animation in CSS. If you could help me with that, that would save me some time with piecing something together.

**Comment by nunocastromartins** [comment]
I simply added this to my `styles.less`
```less
.terminal-cursor {
  -webkit-animation: blink 1s linear infinite;
  @-webkit-keyframes blink {
    50% {
      opacity:0.0;
    }
  }
}
```

**Comment by nunocastromartins** [comment]
Or, with no animation:
```less
.terminal-cursor {
  -webkit-animation: blink steps(1) 1s infinite;
  @-webkit-keyframes blink {
    0% {visibility: visible;}
    50% {visibility: hidden;}
    100% {visibility: visible;}
  }
}
```

**Comment by nunocastromartins** [comment]
With a bit of "hacking" I made it work in a way that blinks without animation, doesn't hide the character in the position of the cursor and depends on the `background-color` of `.terminal-cursor`:

```less
.terminal-cursor {
  position: relative;
  visibility: hidden;
  z-index: 1;
  &:after {
    content: '';
    position: absolute;
    top: 0;
    display: block;
    width: 100%;
    height: 100%;
    background-color: inherit;
    z-index: -1;
    -webkit-animation: blink steps(1) 1s infinite;
    @-webkit-keyframes blink {
      0%   {visibility: visible;}
      50%  {visibility: hidden; }
      100% {visibility: visible;}
    }
  }
}
```

**Comment by jeremyramin** [comment]
I like the animation. However I see the character as hidden when I implement this. Copying that region results in a missing character.
![screen shot 2015-10-30 at 10 32 14 am](https://cloud.githubusercontent.com/assets/3977054/10853087/86a5f77a-7ef1-11e5-9d3d-b2be0ef2ba37.png)
Copying results in: "helo" rather than "hello"

**Comment by nunocastromartins** [comment]
I'm sorry, I edited the comment with the animation too. Forgot to say I edited it, though.
Use `background-color: transparent` and it should work.

**Comment by nunocastromartins** [comment]
I personally prefer the "non-animated" version, as it seems to be the default behaviour in the Atom editor. As a further bonus, changing the `width` in `.terminal-cursor:after` (to, say, 1px or 2px) makes it possible to have an "I-beam" cursor, instead of the "block" one, with no issues (changing the `width` in `.terminal-cursor` causes problems, apparently).

**Comment by jeremyramin** [comment]
Yes, it works now with your updated animation. I added a animation delay to make it blink less often. I could see the cursor being customizable at some point. The only problem with this is that it breaks any theming that people have done in their own stylesheets. I need to make this rely on `.terminal-cursor`'s background color somehow.

**Comment by jeremyramin** [comment]
Turns out, it was a lot simpler than we though:
```less
.terminal-cursor {
        -webkit-animation: blink 1s step-start 0s infinite;
        -webkit-animation-delay: 1s;
        @-webkit-keyframes blink {
          50%  {background: transparent; }
        }
      }
```
This works.

**Comment by nunocastromartins** [comment]
Ah, nice! I'm not an expert on CSS animations. :stuck_out_tongue:
Just note that `animation-delay` is the time until the animation starts, it won't make the animation blink less ofter, I don't think. To make it blink less often just change the time value in the `animation` property.

**Comment by jeremyramin** [comment]
> Ah, nice! I'm not an expert on CSS animations. :stuck_out_tongue:
Neither am I haha.

I should rephrase that. I added the `animation-delay` so the cursor would not consistently blink. The delay will keep the cursor solid during output and while moving the cursor. Then when there is no movement, the cursor will blink. That resembles what most terminal emulators do.

**Comment by jeremyramin** [comment]
Fixed as of v0.12.4. Please give it a try. If you still think this is still an issue, let me know. Otherwise please close this issue if it has been resolved.
